sdk/compiler/daml-lf-proto-encode/src/DA/Daml/LF/Proto3/EncodeV2.hs

```diff
@@ -63,6 +64,17 @@ makeLensesFor [ ("internedKindsMap", "internedKindsMapLens")
               , ("internedTypesMap", "internedTypesMapLens")
               , ("internedExprsMap", "internedExprsMapLens")] ''EncodeEnv
 
+ifSupportsFlatteningM :: Encode a -> Encode a -> Encode a
+ifSupportsFlatteningM = ifM (gets (flip supports featureFlatArchive . version))
+
+ifSupportsFlattening :: a -> a -> Encode a
+ifSupportsFlattening b1 b2 = ifSupportsFlatteningM (return b1) (return b2)
+
+assertSupportsFlattening :: Encode a -> Encode a
+assertSupportsFlattening = flip ifSupportsFlatteningM $ error "assertion failiure: version supports flattening"
```

Comment:
```suggestion
assertSupportsFlattening = flip ifSupportsFlatteningM $ error "assertion failiure: version does not support flattening"
```
I think? Since the flip makes sure this is the false branch, and the function name makes me think this is the intended function.